testutil/trackpr/track.go

```diff
@@ -0,0 +1,269 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//nolint:wrapcheck,cyclop,exhaustruct
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	gh "github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+)
+
+// projectData represents data related to a GitHub project.
+type projectData struct {
+	projectID       gh.ID
+	statusFieldID   gh.ID
+	doneOptionID    gh.String
+	sizeFieldID     gh.ID
+	sprintFieldID   gh.ID
+	currIterationID gh.ID
+}
+
+type PR struct {
+	Title string `json:"title"`
+	Body  string `json:"body"`
+	ID    string `json:"node_id"`
+}
+
+// PRFromEnv returns the PR by parsing it from "GITHUB_PR" env var or an error.
+func PRFromEnv() (PR, error) {
+	const prEnv = "GITHUB_PR"
+	prJSON, ok := os.LookupEnv(prEnv)
+	if !ok || strings.TrimSpace(prJSON) == "" {
+		return PR{}, errors.New("env variable not set", z.Str("var", prEnv))
+	}
+
+	var pr PR
+	if err := json.Unmarshal([]byte(prJSON), &pr); err != nil {
+		return PR{}, errors.Wrap(err, "unmarshal PR body")
+	}
+
+	if pr.Title == "" || pr.Body == "" || pr.ID == "" {
+		return PR{}, errors.New("pr field not set")
+	}
+
+	return pr, nil
+}
+
+// track PR without a ticket and add it to the GitHub project board of the organization.
```

Comment:
`track adds a PR without a ticket to the provided github project setting size=1, iteration=current, status=done`